linker/Linker.Steps/MarkStep.cs

```diff
@@ -811,6 +835,34 @@ bool ProcessLateMarkedAttributes ()
 			return markOccurred;
 		}
 
+		bool ProcessOnlyIfInstantiated ()
+		{
+			// Certain things on a type only need to be marked if an instance of the type could be instantiated
+			var startingQueueCount = _onlyIfInstantiated.Count;
+			if (startingQueueCount == 0)
+				return false;
+			
+			var skippedItems = new List<TypeDefinition> ();
+			var markOccurred = false;
+
+			while (_onlyIfInstantiated.Count != 0) {
+				var type = _onlyIfInstantiated.Dequeue ();
+
+				if (!Annotations.IsInstantiated (type)) {
+					skippedItems.Add (type);
+					continue;
+				}
+				
+				MarkRequirementsForInstantiatedTypes (type);
+			}
+
+			// requeue the items we skipped in case we need to make another pass
+			foreach (var item in skippedItems)
+				_onlyIfInstantiated.Enqueue (item);
+
+			return markOccurred;
```

Comment:
Where is this set?

Reply:
I was able to eliminate this entire method.